core/src/main/java/com/oath/oak/OakDetachedReadKeyBuffer.java

```diff
@@ -6,146 +6,24 @@
 
 package com.oath.oak;
 
-import sun.nio.ch.DirectBuffer;
+import java.util.function.Function;
 
-import java.nio.ByteBuffer;
-import java.nio.ByteOrder;
+class OakDetachedReadKeyBuffer extends OakDetachedReadBuffer {
```

Comment:
I do not get where do we use this buffer in a way that the same ByteBuffer remains cached. It was not in use before this change. And we didn't assume that for the stream iteration that the thread that performs the iteration can not do anything else in parallel.

Reply:
In the old code, we had to update the buffer to set the buffer's position and limit before accessing the buffer.
Here, the detached buffer has an internal attached buffer that updates the position and limit before each access.
So there is no need to "refresh" the buffer.

Reply:
I see now that buffer's limit and position is set on every access. Not a trivial transformation. And we will get read of it in the next PR, right?